src/Compiler/Hoopl/Unique.hs

```diff
@@ -100,8 +100,8 @@ instance IsMap UniqueMap where
 
   mapMap f (UM m) = UM (M.map f m)
   mapMapWithKey f (UM m) = UM (M.mapWithKey (f . intToUnique) m)
-  mapFold k z (UM m) = M.fold k z m
-  mapFoldWithKey k z (UM m) = M.foldWithKey (k . intToUnique) z m
+  mapFold k z (UM m) = M.foldr k z m
+  mapFoldWithKey k z (UM m) = M.foldrWithKey (k . intToUnique) z m
   mapFilter f (UM m) = UM (M.filter f m)
```

Comment:
From the [docs](https://hackage.haskell.org/package/containers-0.5.9.1/docs/Data-Map.html): "This function is an equivalent of foldr and is present for compatibility only."